zygrader/grader.py

```diff
@@ -207,9 +207,10 @@ def run_code_fn(window, submission):
     use_gdb = False
 
     if not submission.compile_and_run_code(use_gdb):
-        popup = ui.layers.OptionsPopup("Error", ["Could not compile code"])
-        popup.add_option("View Log", submission.view_stderr)
-        window.run_layer(popup)
+        if submission.has_stderr():
+            popup = ui.layers.OptionsPopup("Error", ["Could not compile code"])
+            popup.add_option("View Log", submission.view_stderr)
+            window.run_layer(popup)
```

Comment:
It would probably be good to keep and present the popup whether or not stderr is available - only adding the "View Log" if it's available makes sense

I don't think this is strictly related to this PR, but I want to make sure it's addressed and figured might as well comment here.